### ISSUE TITLE
Enable DialogAPI support (and auth) in playground

### ISSUE THREAD

**Issue by Zlatkovsky** [OfficeDev/script-lab#85]
Add helper methods (playground.createDialogUrl() and playground.isDialog()) to allow a snippet to launch in a dialog.

Flow is like this:

**Step 1**: Developer registers their app with AAD or whoever, specifying the Playground Runner as the redirect URL -- no parameters, just plain app.

**Step 2**Developer writes conditional code as follows:

    if (playground.isDialog()) {
        // Assuming this is for auth, check the window.location if the token you want is there.
        if (window.location.href.indexof('code') > 0) {
            Office.context.ui.messageParent("close me");
            return;
        } else {
            window.location.href = "https://login.windows.net/whatever&redirectUrl=" + window.location.href + "...";
            return;
        }
    }

    // otherwise, regular page:
    $('#authenticate').click(authenticate);
    $('#writeDataFromExchange).click(writeDataFromExchange);

    function authenticate() {
        var dialogUrl = playground.createDialogUrl();
        Office.context.ui.displayDialogAsync(dialogUrl, { width: 50, height: 50 }, function (asyncResult) {
            if (asyncResult.status !== Office.AsyncResultStatus.Succeeded) {
                // TODO: Handle error.
                return;
            }
        
            // Get the dialog and register event handlers.
            var dialog = asyncResult.value;
            dialog.addEventHandler(Microsoft.Office.WebExtension.EventType.DialogMessageReceived, function (asyncResult) {
                if (asyncResult.type !== Microsoft.Office.WebExtension.EventType.DialogMessageReceived) {
                    // TODO: Handle unknown message.
                    return;
                }
        
                // We got our data, time to close the dialog.
                dialog.close();
            });
        });
    }
    
    function writeDataFromExchange() {
        // ...
    }

    
The createDialogUrl function creates a URL that is identical to the refresh one, but also add a `isDialog=true` parameter.  This will, in turn, ensure that the the post request also receives this parameter, which makes *it* inject code that will add an entry to SessionStorage, with `playground-is-dialog = true`.  This, in turn, will ensure that `playground.isDialog()` will return `true`.

Note:  we may need to have a different "dialog.html" page that is very similar to refresh.html, but that has the script tag for office.js.  I don't remember if the first page must have that tag in it, or if it just needs to be part of the app domains (which the editor would be at any rate)

**Comment by Zlatkovsky** [comment]
Closing in favor of issue #430.
We do have a current POC prototype, `ScriptLab.getAccessToken`.  But we need to extend it to use OfficeJsHelpers before we can ship it, and we also want to be able to register on-the-fly with AAD.
